common/xmlparser.cpp

```diff
@@ -340,8 +340,24 @@ bool XMLParser::parse() {
 		case kParserNeedHeader:
 		case kParserNeedKey:
 			if (_char != '<') {
-				parserError("Parser expecting key start.");
-				break;
+				if (_allowText) {
+					Common::String text;
+					while (_char != '<' && _char) {
```

Comment:
a `do { } while()` would be more logical here

Reply:
True. For some reason it never occurs to me to use one, but it is better here.